SayTheirNames/Source/Controller/Petitions/PetitionsController.swift

```diff
@@ -9,20 +9,84 @@
+        
+        dataSource.findOutMoreAction = { [weak self] petition in
+            
+            self?.showDetailViewController(for: petition)
+        }
+        return dataSource
+    }()
     
     required init(service: Service) {
-        super.init(service: service)
+        self.service = service
+        super.init(nibName: nil, bundle: nil)
     }
     
-    @available(*, unavailable)
-    required init?(coder: NSCoder) {
-        fatalError()
+    required init?(coder: NSCoder) { fatalError("This should not be called") }
+
+    override func loadView() {
+        self.view = petitionsView
     }
     
-    override func loadView() {
-        view = viewCode
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        petitionsTableView.delegate = self
+        
+        // build mock data for display now
+        // in the future, we'll get petitions from the network
+        let longTitle = "Editing the new edition of revolution"
+        let shortTitle = "Justice for George Floyd"
+        
+        let dummyPetitions = [
+            MockPetition(title: longTitle, verified: true, hasImage: true),
+            MockPetition(title: longTitle, verified: false, hasImage: false),
+            MockPetition(title: longTitle, verified: true, hasImage: true),
+            MockPetition(title: longTitle, verified: true, hasImage: false),
+            MockPetition(title: shortTitle, verified: true, hasImage: true),
+            MockPetition(title: shortTitle, verified: true, hasImage: false),
+            MockPetition(title: shortTitle, verified: false, hasImage: true),
+            MockPetition(title: shortTitle, verified: true, hasImage: false)
+        ]
+        
+        dataSource.set(petitions: dummyPetitions)
+    }
+        
+    private func showDetailViewController(for petition: PresentedPetition) {
+        guard let service = service else { return }
+        
+        // People CollectionView
+        // let selectedPerson = peopleDataSource.fetchPerson(at: indexPath.item)
+        let detailViewController = PetitionDetailViewController(service: service, petition: petition)
+        
+        let navigationController = UINavigationController(rootViewController: detailViewController)
+        navigationController.navigationBar.isHidden = true
+        present(navigationController, animated: true, completion: nil)
+    }
+}
+
+// MARK: - UITableViewDelegate
+extension PetitionsController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        // there's no real selected state in this design,
+        // we simply show another view controller when the find out more button is tapped
+        false
     }
 }
+
+// MARK: - Constants
+extension PetitionsController {
+    
+    static let PetitionTitle = NSLocalizedString("Petitions", comment: "Petitions View Controller Title")
```

Comment:
We have SwiftGen integrated and there should be something in `Strings` enum. fell free to leave a to-do though (or a trello card)

Reply:
I saw that it was being considered, didn't realize it was in the development branch yet.  Done.